core/drivers/imx_snvs.c

```diff
@@ -1,186 +1,52 @@
 // SPDX-License-Identifier: BSD-2-Clause
 /*
- * Copyright 2017 NXP
- * Copyright (c) 2014, 2015 Linaro Limited
- *
- * Peng Fan <peng.fan@nxp.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *
- * 1. Redistributions of source code must retain the above copyright notice,
- * this list of conditions and the following disclaimer.
- *
- * 2. Redistributions in binary form must reproduce the above copyright notice,
- * this list of conditions and the following disclaimer in the documentation
- * and/or other materials provided with the distribution.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
+ * Copyright (C) 2020 Pengutronix
+ * Rouven Czerwinski <entwicklung@pengutronix.de>
  */
 
-#include <initcall.h>
 #include <io.h>
 #include <drivers/imx_snvs.h>
-#include <kernel/misc.h>
-#include <kernel/tee_time.h>
-#include <kernel/time_source.h>
 #include <mm/core_memprot.h>
 #include <mm/core_mmu.h>
-#include <mpa.h>
-#include <utee_defines.h>
-#include <platform_config.h>
 #include <stdint.h>
-#include <tee/tee_cryp_utl.h>
 #include <types_ext.h>
 #include <trace.h>
 
-#define SNVS_LPLR	0x34
-#define SNVS_LPCR	0x38
-#define SNVS_LPMKCR	0x3C
-#define SNVS_LPSVCR	0x40
-#define SNVS_LPTGFCR	0x44
-#define SNVS_LPTDCR	0x48
-#define SNVS_LPSR	0x4C
-#define SNVS_LPSRTCMR	0x50
-#define SNVS_LPSRTCLR	0x54
-#define SNVS_LPTAR	0x58
-#define SNVS_LPSMCMR	0x5C
-#define SNVS_LPSMCLR	0x60
-#define SNVS_LPPGDR	0x64
-#define SNVS_LPGPR0_A	0x68
-#define SNVS_LPZMKR0	0x6C
-#define SNVS_LPCGR0_30	0x90
-#define SNVS_LPCGR0_31	0x94
-#define SNVS_LPCGR0_32	0x98
-#define SNVS_LPCGR0_33	0x9C
-#define SNVS_LPTDC2R	0xA0
-#define SNVS_LPTDSR	0xA4
-#define SNVS_LPTGF1CR	0xA8
-#define SNVS_LPTGF2CR	0xAC
-#define SNVS_LPAT1CR	0xC0
-#define SNVS_LPAT2CR	0xC4
-#define SNVS_LPAT3CR	0xC8
-#define SNVS_LPAT4CR	0xCC
-#define SNVS_LPAT5CR	0xD0
-#define SNVS_LPATCTLR	0xE0
-#define SNVS_LPATCLKR	0xE4
-#define SNVS_LPATRC1R	0xE8
-#define SNVS_LPATRC2R	0xEC
+#define SNVS_HPSR	0x34
```

Comment:
this value (SNVS_HPSR) should be 0x14 (not 0x34) ..at least that is what I find in imx7ulp

Reply:
Yes, I fixed this in the downstream branch, I'll fix this while reworking.